x-pack/plugins/enterprise_search/public/applications/workplace_search/components/overview/overview.tsx

```diff
@@ -0,0 +1,152 @@
+import { KibanaContext, IKibanaContext } from '../../../index';
+
+import { IAccount } from '../../types';
+
+import { ErrorState } from '../error_state';
+
+import { Loading } from '../shared/loading';
+import { ProductButton } from '../shared/product_button';
+import { ViewContentHeader } from '../shared/view_content_header';
+
+import { OnboardingSteps } from './onboarding_steps';
+import { OrganizationStats } from './organization_stats';
+import { RecentActivity, IFeedActivity } from './recent_activity';
+
+export interface IAppServerData {
+  hasUsers: boolean;
+  hasOrgSources: boolean;
+  canCreateContentSources: boolean;
+  canCreateInvitations: boolean;
+  isOldAccount: boolean;
+  sourcesCount: number;
+  pendingInvitationsCount: number;
+  accountsCount: number;
+  personalSourcesCount: number;
+  activityFeed: IFeedActivity[];
+  organization: {
+    name: string;
+    defaultOrgName: string;
+  };
+  isFederatedAuth: boolean;
+  currentUser: {
+    firstName: string;
+    email: string;
+    name: string;
+    color: string;
+  };
+  fpAccount: IAccount;
+}
+
+export const defaultServerData = {
+  accountsCount: 1,
+  activityFeed: [],
+  canCreateContentSources: true,
+  canCreateInvitations: true,
+  currentUser: {
+    firstName: '',
+    email: '',
+    name: '',
+    color: '',
+  },
+  fpAccount: {} as IAccount,
+  hasOrgSources: false,
+  hasUsers: false,
+  isFederatedAuth: true,
+  isOldAccount: false,
+  organization: {
+    name: '',
+    defaultOrgName: '',
+  },
+  pendingInvitationsCount: 0,
+  personalSourcesCount: 0,
+  sourcesCount: 0,
+  telemetryStatus: {},
+} as IAppServerData;
+
+const ONBOARDING_HEADER_TITLE = i18n.translate(
+  'xpack.enterpriseSearch.workplaceSearch.overviewOnboardingHeader.title',
+  { defaultMessage: 'Get started with Workplace Search' }
+);
+
+const HEADER_TITLE = i18n.translate('xpack.enterpriseSearch.workplaceSearch.overviewHeader.title', {
+  defaultMessage: 'Organization overview',
+});
+
+const ONBOARDING_HEADER_DESCRIPTION = i18n.translate(
+  'xpack.enterpriseSearch.workplaceSearch.overviewOnboardingHeader.description',
+  { defaultMessage: 'Complete the following to set up your organization.' }
+);
+
+const HEADER_DESCRIPTION = i18n.translate(
+  'xpack.enterpriseSearch.workplaceSearch.overviewHeader.description',
+  { defaultMessage: "Your organizations's statistics and activity" }
+);
+
+export const Overview: React.FC = () => {
+  const { http } = useContext(KibanaContext) as IKibanaContext;
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasErrorConnecting, setHasErrorConnecting] = useState(false);
+  const [appData, setAppData] = useState(defaultServerData);
+
+  const getAppData = async () => {
+    try {
+      const response = await http.get('/api/workplace_search/overview');
+      setAppData(response);
+      setIsLoading(false);
+    } catch (error) {
+      setHasErrorConnecting(true);
+      setIsLoading(false);
```

Comment:
We can move `setIsLoading(false);` to a `finally {}` block if we're using it in both try/catch

Reply:
Addressed in cdcff72